### PR TITLE
Add ecoCode builtin profile for ecocode-java plugin

### PR DESCRIPTION
**Pull request by jycr**
This makes it easier to implement #53

### CODE REVIEW

File: src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	public void define(Context context) {
+		NewBuiltInQualityProfile ecoCodeProfile = context.createBuiltInQualityProfile(PROFILE_NAME, LANGUAGE);
+		loadProfile(ecoCodeProfile);
+		ecoCodeProfile.done();
+	}
+
+	void loadProfile(NewBuiltInQualityProfile profile) {
+		BuiltInQualityProfileJsonLoader.load(profile, REPOSITORY_KEY, PROFILE_PATH);
```
Comment (dedece35) [inline src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java:38]: what about setting this new profile as default profile for java language ? 
is it wanted that it isn't the default profile for java language ?

File: src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+package fr.greencodeinitiative.java;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonarsource.analyzer.commons.BuiltInQualityProfileJsonLoader;
+
+import static fr.greencodeinitiative.java.JavaRulesDefinition.LANGUAGE;
+import static fr.greencodeinitiative.java.JavaRulesDefinition.REPOSITORY_KEY;
+
+public final class JavaEcoCodeProfile implements BuiltInQualityProfilesDefinition {
```
Comment (dedece35) [inline src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java:26]: when is this profile created ? automatically, when the developer will launch unit tests ?
what about conflicts with developers using docker method ?

File: src/main/resources/fr/greencodeinitiative/java/ecocode_profile.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "ecoCode",
+  "ruleKeys": [
```
Comment (dedece35) [inline src/main/resources/fr/greencodeinitiative/java/ecocode_profile.json:3]: this list is static. could we use a dynamic list got from code ?

**Review by dedece35 — CHANGES_REQUESTED** [review]

File: src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java (hunk 2 above)
Comment (jycr) [reply src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java:26]: We should not force the change of default profile, but simply propose a profile grouping all the ecoCode rules in order to facilitate their application for those who might be interested.
The presence of this profile will help with automated integration tests

File: src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java (hunk 1 above)
Comment (jycr) [reply src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java:38]: We should not force the change of default profile, but simply propose a profile grouping all the ecoCode rules in order to facilitate their application for those who might be interested.
The presence of this profile will help with automated integration tests

File: src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java (hunk 2 above)
Comment (jycr) [reply src/main/java/fr/greencodeinitiative/java/JavaEcoCodeProfile.java:26]: This profile is automatically created when plugin is installed.

File: src/main/resources/fr/greencodeinitiative/java/ecocode_profile.json (hunk 3 above)
Comment (jycr) [reply src/main/resources/fr/greencodeinitiative/java/ecocode_profile.json:3]: We could create a JSON file at compile time.
But for now, we can start by testing if the profile contains all the rules implemented by this plugin.
We can improve this in a next PR?

**Review by dedece35 — APPROVED** [review]

**Pull request merged**